x-pack/plugins/observability_solution/infra/public/alerting/metric_threshold/rule_data_formatters.ts

```diff
@@ -5,14 +5,41 @@
  * 2.0.
  */
 
-import { ALERT_REASON } from '@kbn/rule-data-utils';
+import { ALERT_REASON, ALERT_RULE_PARAMETERS } from '@kbn/rule-data-utils';
 import { ObservabilityRuleTypeFormatter } from '@kbn/observability-plugin/public';
+import { LocatorPublic } from '@kbn/share-plugin/common';
+import type { AssetDetailsLocatorParams } from '@kbn/observability-shared-plugin/common';
+import { InventoryItemType, findInventoryModel } from '@kbn/metrics-data-access-plugin/common';
+import { SupportedAssetTypes } from '../../../common/asset_details/types';
 import { getMetricsViewInAppUrl } from '../../../common/alerting/metrics/alert_link';
-export const formatReason: ObservabilityRuleTypeFormatter = ({ fields }) => {
-  const reason = fields[ALERT_REASON] ?? '-';
 
-  return {
-    reason,
-    link: getMetricsViewInAppUrl(fields),
+export const getFormatReason = ({
+  assetDetailsLocator,
+}: {
+  assetDetailsLocator?: LocatorPublic<AssetDetailsLocatorParams>;
+}): ObservabilityRuleTypeFormatter => {
+  return ({ fields }) => {
+    const reason = fields[ALERT_REASON] ?? '-';
+    const groupBy = (fields[ALERT_RULE_PARAMETERS]?.groupBy as string[]) ?? [];
+
+    const assetTypeByAssetId = Object.values(SupportedAssetTypes).reduce((acc, curr) => {
+      acc[findInventoryModel(curr).fields.id] = curr;
+      return acc;
+    }, {} as Record<string, InventoryItemType>);
+
+    const supportedAssetId = groupBy.find((field) => !!assetTypeByAssetId[field]);
+    const assetType = supportedAssetId ? assetTypeByAssetId[supportedAssetId] : undefined;
+
+    const locator = assetType ? assetDetailsLocator : undefined;
+
+    return {
+      reason,
+      link: getMetricsViewInAppUrl({
+        fields,
+        assetDetailsLocator: locator,
+        nodeType: assetType,
+      }),
+      hasBasePath: !!assetType,
```

Comment:
Why do we need to set this parameter?

Reply:
This is something used by the [`alert-actions`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/observability/public/pages/alerts/components/alert_actions.tsx#L87). I think it's a way for it to work with support URLs generated by locators.